clients/include/testing_trsm_batched.hpp

```diff
@@ -0,0 +1,403 @@
+                    if(hX[b][idx] != 0)
+                    {
+                        err_1 += std::abs((hX[b][idx] - hXorB_1[b][idx]) / hX[b][idx]);
+                        err_2 += std::abs((hX[b][idx] - hXorB_2[b][idx]) / hX[b][idx]);
+                    }
+                    else
+                    {
+                        err_1 += std::abs(hXorB_1[b][idx]);
+                        err_2 += std::abs(hXorB_2[b][idx]);
+                    }
+                }
+                max_err_1 = max_err_1 > err_1 ? max_err_1 : err_1;
+                max_err_2 = max_err_2 > err_2 ? max_err_2 : err_2;
+            }
+
+            trsm_err_res_check<T>(max_err_1, M, error_eps_multiplier, eps);
+            trsm_err_res_check<T>(max_err_2, M, error_eps_multiplier, eps);
+        }
+
+        // Residual Check
+        // hXorB <- hA * (A^(-1) B) ;
+        for(int b = 0; b < batch_count; b++)
+        {
+            cblas_trmm<T>(
+                side, uplo, transA, diag, M, N, 1.0 / alpha_h, hA[b], lda, hXorB_1[b], ldb);
+            cblas_trmm<T>(
+                side, uplo, transA, diag, M, N, 1.0 / alpha_h, hA[b], lda, hXorB_2[b], ldb);
+        }
+
+        // hXorB contains A * (calculated X), so residual = A * (calculated X) - B
+        //                                                = hXorB - hB
+        // res is the one norm of the scaled residual for each column
+        for(int b = 0; b < batch_count; b++)
+        {
+            max_res_1 = max_res_2 = 0;
+            for(int i = 0; i < N; i++)
+            {
+                T res_1 = 0.0;
+                T res_2 = 0.0;
+                for(int j = 0; j < M; j++)
+                {
+                    int idx = j + i * ldb;
+                    if(hB[b][j + i * ldb] != 0)
+                    {
+                        res_1 += std::abs((hXorB_1[b][idx] - hB[b][idx]) / hB[b][idx]);
+                        res_2 += std::abs((hXorB_2[b][idx] - hB[b][idx]) / hB[b][idx]);
+                    }
+                    else
+                    {
+                        res_1 += std::abs(hXorB_1[b][idx]);
+                        res_2 += std::abs(hXorB_2[b][idx]);
+                    }
+                }
+                max_res_1 = max_res_1 > res_1 ? max_res_1 : res_1;
+                max_res_2 = max_res_2 > res_2 ? max_res_2 : res_2;
+            }
+            trsm_err_res_check<T>(max_res_1, M, residual_eps_multiplier, eps);
+            trsm_err_res_check<T>(max_res_2, M, residual_eps_multiplier, eps);
+        }
+    }
+
+    if(arg.timing)
+    {
+        double gpu_time_used, cpu_time_used;
+        double rocblas_gflops, cblas_gflops;
+
+        // GPU rocBLAS
+        for(int b = 0; b < batch_count; b++)
+        {
+            CHECK_HIP_ERROR(
+                hipMemcpy(XorBv[b], hXorB_1[b], sizeof(T) * size_B, hipMemcpyHostToDevice));
+        }
+        CHECK_HIP_ERROR(hipMemcpy(dXorB, hXorB_1, sizeof(T*) * batch_count, hipMemcpyHostToDevice));
```

Comment:
I think hXorB_1 should be XorBv

Reply:
All good points. Correcting them all, should all be fixed in the next commit, just waiting for tensile to build to make sure I didn't break anything.

Reply:
Fixed in 9d378b0.